Python/optimizer_analysis.c

```diff
@@ -617,7 +618,7 @@ remove_unneeded_uops(_PyUOpInstruction *buffer, int buffer_size)
                     while (op_skip[last->opcode]) {
                         last--;
                     }
-                    if (op_without_push[last->opcode]) {
+                    if (op_without_push[last->opcode] && op_without_pop[opcode]) {
                         last->opcode = op_without_push[last->opcode];
                         opcode = buffer[pc].opcode = op_without_pop[opcode];
```

Comment:
When we are here, we know that `op_without_pop[opcode] || op_without_pop_null[opcode]` is true. But we should also make sure that `op_without_pop[opcode]` itself is true, otherwise it is possible to accidentally write `0` as the opcode.